types/rosie/index.d.ts

```diff
@@ -14,7 +14,7 @@ declare namespace rosie {
      * @param {function(object): *=} constructor
      * @return {Factory}
      */
-    define<T = any>(name: string, constructor?: (opts?: any) => any): IFactory<T>;
+    define<T = any>(name: string, constructor?: any): IFactory<T>;
```

Comment:
Could the type be `((opts?: any) => any) | (new(opts?: any) => any))`?

Reply:
Hey Andy! I was unclear on how to express it and what was there wasn't working, this is an attempt to at least let it compile and preserve functionality. Is that the syntax for a an object that has a constructor?

Reply:
Yep, `new(opts?: any) => any` is equivalent to `{ new(opts?: any): any }`.

Reply:
Ah, that's really cool. I'll update this when I get a chance, probably tomorrow. Thank you!